harness/determined/pytorch/deepspeed/_deepspeed_trial.py

```diff
@@ -0,0 +1,881 @@
+import abc
+import logging
+import pathlib
+import pickle
+import random
+import time
+from typing import Any, Dict, Iterator, List, Optional, Type, Union, cast
+
+import deepspeed
+import numpy as np
+import torch
+
+import determined as det
+from determined import layers, pytorch, util, workload
+from determined.common import storage
+from determined.pytorch import _metric_utils
```

Comment:
can you expose the things you need from `_metric_utils` in `pytorch/__init__.py` to avoid this extra module?  This case is [explained in detail in the code layout rfc](https://github.com/rb-determined-ai/pedl/blob/9bd2168ec23fbba866194f3d7d3cfc243d267100/python_code_layout.md#special-case-handling-private-functions-and-classes-in-private-modules), if you are interested in the long-form explanation.

Exposing the methods within `pytorch._metric_utils` directly in `pytorch` makes the code easier to work with, since you have one less thing to import.  It makes it easier to refactor the code because if you move those functions, then you change how they get imported into the `pytorch` namespace but the rest of the code is unaffected (no having to rename `_metric_utils` everywhere, for instance).  It doesn't affect user code performance because the `_metrics_utils` will be imported automatically anyway by `_pytorch_trial`.  And it doesn't affect user namespaces because all of the methods in question are already private methods.

Reply:
thanks for the detailed explanation.  i've refactored `PyTorchTrial` to call functions from `_metric_utils.py`.